modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java

```diff
@@ -242,18 +241,15 @@ public void testDeleteByQueryOnReadOnlyAllowDeleteIndex() throws Exception {
             // it will trigger a retry policy in the delete by query request because the rest status of the block is 429
             enableIndexBlock("test", SETTING_READ_ONLY_ALLOW_DELETE);
             if (diskAllocationDeciderEnabled) {
-                InternalTestCluster internalTestCluster = internalCluster();
-                InternalClusterInfoService infoService = (InternalClusterInfoService) internalTestCluster
-                    .getInstance(ClusterInfoService.class, internalTestCluster.getMasterName());
-                ThreadPool threadPool = internalTestCluster.getInstance(ThreadPool.class, internalTestCluster.getMasterName());
-                // Refresh the cluster info after a random delay to check the disk threshold and release the block on the index
-                threadPool.schedule(
-                        () -> ClusterInfoServiceUtils.refresh(infoService),
-                        TimeValue.timeValueMillis(randomIntBetween(1, 100)),
-                        ThreadPool.Names.MANAGEMENT);
-                // The delete by query request will be executed successfully because the block will be released
-                assertThat(deleteByQuery().source("test").filter(QueryBuilders.matchAllQuery()).refresh(true).get(),
-                    matcher().deleted(docs));
+                // Fire off the delete-by-query first
```

Comment:
I like this simplification, so just out of curiosity, did not changing this cause a failure? I need a hint to realize how I think.

Reply:
Yes, the `threadPool.schedule(..., ThreadPool.Names.MANAGEMENT)` made a blocking call on the (sole) management thread that could not complete because the refresh needs to make stats calls which also need management threads. I did a targeted fix in b8a20ddd4cb84352ef947c5cd3e0378e6ffd5135 but then decided that the whole thing could be simplified.

Reply:
Thanks, got it now.